### PR TITLE
integrate jvm.options template and default variables; #222

### LINKED ISSUE #222: Improve jvm.option handling with dynamic
The actuall way to set memory heapsize is very simple and formless.
Actuallly it is also not possible to set any kindo of custom jvm parameters.

There cases where it is helpful and/or needed to use custom parameters. Best example is if you are forced to work behind proxy servers and then native java bundle whithin elasticsearch must be able to reach external servers (e.g GeoIP database updates)

### PR DESCRIPTION
**Pull request by xtruthx**
This PR  is for #222

Additional Documenation will be contributed later on. I don't wont to contribute separatly into documentation. It will be done with the contribution for the review.

fixes #222

### CODE REVIEW

**Review by widhalmt — APPROVED** [review]
Thank you very much for the contribution. I didn't like the "quick hack" with the `lineinfile` anyway. Having it as a template is defnitely the better way.

**Review by widhalmt — CHANGES_REQUESTED** [review]
Damn, missed the variable naming. I'll add a suggestion.

File: roles/elasticsearch/defaults/main.yml [hunk 1]
```diff
@@ -18,6 +18,15 @@ elasticsearch_heap: "{{ [[(ansible_memtotal_mb // 1024) // 2, 30] | min, 1] | ma
...
 elasticsearch_clustername: elasticsearch
+elasticsearch_conf_dir: "/etc/elasticsearch/"
+elasticsearch_config_jvm: "jvm.options.j2"
+elasticsearch_user: elasticsearch
+elasticsearch_group: elasticsearch
+
+# JVM custom parameters
+es_java_home: ''
+es_jvm_custom_parameters: ''
+es_heap_dump_path: "/var/lib/elasticsearch"
```
Comment (widhalmt) [inline roles/elasticsearch/defaults/main.yml:29]:
```suggestion
elasticsearch_java_home: ''
elasticsearch_jvm_custom_parameters: ''
elasticsearch_heap_dump_path: "/var/lib/elasticsearch"
```

File: roles/elasticsearch/templates/jvm.options.j2 [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+## -Xmx4g
+##
+## See https://www.elastic.co/guide/en/elasticsearch/reference/current/heap-size.html
+## for more information
+##
+################################################################
+
+# Xms represents the initial size of total heap space
+# Xmx represents the maximum size of total heap space
+{% if es_heap_size is defined %}
```
Comment (widhalmt) [inline roles/elasticsearch/templates/jvm.options.j2:21]:
```suggestion
{% if elasticsearch_heap_size is defined %}
```

File: roles/elasticsearch/templates/jvm.options.j2 [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+8:-XX:+UseGCLogFileRotation
+8:-XX:NumberOfGCLogFiles=32
+8:-XX:GCLogFileSize=64m
+
+# JDK 9+ GC logging
+9-:-Xlog:gc*,gc+age=trace,safepoint:file={{ elasticsearch_logpath }}/gc.log:utctime,pid,tags:filecount=32,filesize=64m
+{% endif %}
+
+{% if es_jvm_custom_parameters !='' %}
+{% for item in es_jvm_custom_parameters %}
```
Comment (widhalmt) [inline roles/elasticsearch/templates/jvm.options.j2:146]:
```suggestion
{% if elasticsearch_jvm_custom_parameters !='' %}
{% for item in elasticsearch_jvm_custom_parameters %}
```

File: roles/elasticsearch/templates/jvm.options.j2 [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+## heap dumps
+
+# generate a heap dump when an allocation from the Java heap fails
+# heap dumps are created in the working directory of the JVM
+-XX:+HeapDumpOnOutOfMemoryError
+
+# specify an alternative path for heap dumps; ensure the directory exists and
+# has sufficient space
+-XX:HeapDumpPath={{ es_heap_dump_path }}
```
Comment (widhalmt) [inline roles/elasticsearch/templates/jvm.options.j2:125]:
```suggestion
-XX:HeapDumpPath={{ elasticsearch_heap_dump_path }}
```

**Review by widhalmt — CHANGES_REQUESTED** [review]
Our naming scheme for variables says: "role name for role specific variables. And collection name for 'global' variables". This had to be adjusted to fulfill new linter rules.

**Comment by xtruthx** [comment]
suggestions accepted and done

**Review by dnssmr — APPROVED** [review]
Looks good to me

**Review by widhalmt — APPROVED** [review]

**Pull request merged**